go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -465,16 +471,17 @@ func (n *OvnNode) initSharedGateway(subnets []*net.IPNet, gwNextHop net.IP, gwIn
 		bridgeName = gwIntf
 	}
 
-	// Now, we get IP address from OVS bridge. If IP does not exist,
+	// Now, we get IP addresses from OVS bridge. If IP does not exist,
 	// error out.
-	ipAddress, err := getIPv4Address(gwIntf)
+	ips, err := getNetworkInterfaceIPAddresses(gwIntf)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get interface details for %s (%v)",
 			gwIntf, err)
 	}
-	if ipAddress == nil {
-		return nil, fmt.Errorf("%s does not have a ipv4 address", gwIntf)
+	if len(ips) == 0 {
+		return nil, fmt.Errorf("%s does not have an IP address", gwIntf)
```

Comment:
Need to make sure we have an IP address corresponding to each family in `subnets`

Reply:
Moved error handling to within getNetworkInterfaceIPAddresses().
Done.